runtime/lua/vim/lsp/util.lua

```diff
@@ -1991,6 +1991,7 @@ end
 ---
 --- CAUTION: Modifies the input in-place!
 ---
+---@deprecated
 ---@param lines table list of lines
 ---@return string filetype or "markdown" if it was unchanged.
```

Comment:
If the deprecated functions are not used in this codebase then we can go ahead and delete their docs too.
```suggestion
---@deprecated
```

Reply:
I prefer to leave the docs for now and delete them together with the code in the future. The type annotations also help LuaLS (type warnings give me anxiety).

Reply:
For the sake of completeness: We _could_ keep just the annotations and remove the descriptions, but I agree that while the code is there, the full documentation is useful for code archeology (e.g., no need to hunt further for the commit _with_ description).